inc/saiswitch.h

```diff
@@ -2776,6 +2776,30 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_ECMP_MEMBER_COUNT,
 
+    /**
+     * @brief Enable or disable credit watchdog
+     *
+     * Credit Watchdog can be enabled or disabled using this attribute for VOQ based system
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default true
+     * @validonly SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_VOQ
+     */
+    SAI_SWITCH_ATTR_CREDIT_WD,
+
+    /**
+     * @brief Credit watchdog threshold timer in milliseconds
+     * Value must be within 10ms - 1000ms range
+     * Queue is set to delete state and all packets in queue are deleted after the timer expiry
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 500
+     * @validonly SAI_SWITCH_ATTR_TYPE == SAI_SWITCH_TYPE_VOQ
+     */
+    SAI_SWITCH_ATTR_CREDIT_WD_TIMER,
```

Comment:
good to add counters where packets got deleted due to this.

Reply:
@JaiOCP are we adding a counter for this? Thanks.

Reply:
Added the counter for tracking packet deletes because of credit WD expiry.
https://github.com/opencomputeproject/SAI/pull/1748/commits/dd2e192b4b75fd07d7a8088aeccbf620e5e00423